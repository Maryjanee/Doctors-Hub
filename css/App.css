.form-container {
  background: white;
  width: 85%;
  margin: 0 auto;
  top: 50%;
  left: 50%;
  -webkit-transform: translate3d(-50%, -50%, 0);
          transform: translate3d(-50%, -50%, 0);
  position: absolute;
  padding: 1.5rem 2rem;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.form-container input,
.form-container button[type="submit"] {
  padding: 1rem 1.2rem;
  margin: 1rem 0;
  border: 0;
  width: 100%;
  font-size: 1.15rem;
  background: #f2f2f2;
  color: black;
}

.form-container input :focus,
.form-container button[type="submit"] :focus {
  outline: 2px solid #ffb400;
}

.form-container .create {
  color: #ffb400;
  margin-left: 0.5rem;
}

@media (min-width: 768px) {
  .form-container {
    width: 35%;
  }
  .form-container input[type="email"]:focus,
  .form-container input[type="text"]:focus,
  .form-container input[type="password"]:focus {
    outline: 2px solid #ffb400;
  }
}
/*# sourceMappingURL=App.css.map */