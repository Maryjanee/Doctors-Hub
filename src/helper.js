/* eslint-disable import/prefer-default-export */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable arrow-body-style */
import { BrowserRouter } from 'react-router-dom';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import rootReducer from './reducers';

export const renderComponent = (Component, props = {}) => {
  return (
    <BrowserRouter>
      <Component {...props} />
    </BrowserRouter>
  );
};

export const renderWithRedux = (component,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  } = {}) => ({
  ...render(
    <Provider store={store}>
      renderComponent(component)
    </Provider>,
  ),
});
