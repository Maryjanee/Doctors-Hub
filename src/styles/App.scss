$yellow: #ffb400;
$padding: 1rem 1.5rem;
$white: #fff;
$grey: #cacaca;

.form-container {
  background: white;
  width: 85%;
  margin: 0 auto;
  top: 50%;
  left: 50%;
  transform: translate3d(-50%, -50%, 0);
  position: absolute;
  padding: 1.5rem 2rem;
  justify-content: center;

  input {
    background: #f2f2f2;
  }

  button[type="submit"] {
    background: $yellow !important;
  }

  input,
  button[type="submit"] {
    padding: 1rem 1.2rem;
    margin: 1rem 0;
    border: 0;
    width: 100%;
    font-size: 1.15rem;
    color: black;

    :focus {
      outline: 2px solid $yellow;
    }
  }

  .create {
    color: $yellow;
    margin-left: 0.5rem;
  }

  @media (min-width: 768px) {
    width: 35%;

    input[type="email"]:focus,
    input[type="text"]:focus,
    input[type="password"]:focus {
      outline: 2px solid $yellow;
    }
  }
}
