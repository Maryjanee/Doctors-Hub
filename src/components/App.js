import { useEffect } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import Login from './Login';
import SignUp from './Signup';
import Doctors from '../containers/Doctors';
import Doctor from '../containers/Doctor';
import { fetchDoctors } from '../actions/index';
import ProtectedRoute from './ProtectedRoute';
import Appointment from './Appointments';
import FourOhFour from './404';
import '../styles/App.scss';

const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    if (localStorage.getItem('auth-token')) {
      dispatch(fetchDoctors());
    }
  }, []);
  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/" component={Login} />
        <Route path="/signup" component={SignUp} />
        <ProtectedRoute exact path="/doctors" component={Doctors} />
        <ProtectedRoute path="/doctor/:doctorId" component={Doctor} />
        <ProtectedRoute exact path="/appointments" component={Appointment} />
        <Route path="/*" component={FourOhFour} />
      </Switch>
    </BrowserRouter>
  );
};

export default App;
