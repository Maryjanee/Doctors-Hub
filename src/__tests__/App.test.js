import App from '../components/App';
import { renderWithRedux } from './helper';

describe('App Component', () => {
  it('renders with redux and displays h1 text', () => {
    const { getByText } = renderWithRedux(<App />);
    expect(getByText('Doctors Hub')).toBeInTheDocument();
  });
});
