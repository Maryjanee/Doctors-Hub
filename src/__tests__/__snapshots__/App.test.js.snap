// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login page correctly 1`] = `
<div
  className="main-bg"
>
  <div
    className="container"
  >
    <h1
      className="logo"
    >
      Doctors Hub
    </h1>
    <div
      className="form-container"
    >
      <form
        className="d-flex flex-cl"
        onSubmit={[Function]}
      >
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <input
          className="submit"
          type="submit"
          value="Login"
        />
      </form>
      <p
        className="text-center"
      >
        Not Registered?
        <a
          className="create"
          href="/signup"
          onClick={[Function]}
        >
          Create an Account
        </a>
      </p>
    </div>
  </div>
</div>
`;
